batch_test.go

```diff
@@ -26,7 +26,7 @@ func (d *testDag) Get(ctx context.Context, cid cid.Cid) (Node, error) {
 	if n, ok := d.nodes[cid.KeyString()]; ok {
 		return n, nil
 	}
-	return nil, ErrNotFound{cid}
+	return nil, ErrNotFound{Key: cid.KeyString()}
```

Comment:
> This is a BREAKING CHANGE as it no longer strictly matches only this ErrNotFound type but any type implementing interface{ NotFound() bool }.

It's also breaks here because the field and value of the error have changed (note: 

What's the upgrade plan here for all the impacted repos if these PRs were to get merged? Is it easier/harder than:
1. Just having go-ipld-prime's error type alias this one (since IIUC you introducing a new type for go-ipld-prime as opposed to this one which is already in use)
2. Making a new type in go-ipld-prime, adding the `NotFound() bool` detection to both types and fixing any stragglers comparing directly with this type rather than using a function for determining equality?

Is the assumption here that there are more places doing equality checks (or type casts) vs creating or getting data from the error type? That the compiler will be louder here which will make this easier, or something else?

Reply:
1. The original impetus behind this exercise is to remove dependency on go-ipld-format, because as far as a large amount of new code is concerned this repo only exists for this error type. I think there is a path to _eventually_ deprecating this repo, but this error type will remain the main blocker to that kind of cleanup.
2. That was my original proposal, but the objection (further up here) was that `errors.Is` should be the preferred method of error checking, operating on a single, sentinel error type. Finding a single solution is a bit challenging.

I don't think I've seen any code directly doing type or equality checks with this type. I may not be looking in the right places. The vast majority of uses call directly into `IsNotFound`, which we can keep stable easily enough. (From a GitHub search that there's ~3 places (not forks) using `errors.Is()`).

Yet another possibility here is closer to my original commit here: Don't alias the error type, leave it alone, but make the implementations here and in go-ipld-prime roughly the same (probably with different field names) - performing the same feature test. As we move dependent code forward, both `errors.Is()` and `IsNotFound()` will pick up both forms, just using the idea that a "not found error conforms to a strict contract". That option just removes the singular sentinel form, which was requested above but I'm fine with that because it gives us all room to move I think.